Convert MetOffice to use UI for configuration

MrHarcombe opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->
As this moves the config for the MetOffice weather integration into the UI from the YAML, you will need to ensure you keep a copy of the DataPoint API key handy to re-enable the integration in your server. The location being forecast is, by default, taken as the location of the server, so again if you were forecasting other locations you will need to have those GPS co-ordinates to hand, to re-enter them when configuring new integrations through the UI.

## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
In line with the request made following PR 34642 (link below), this change moves the MetOffice weather integration away from YAML to use the UI (config_flow) configuration. 

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: https://github.com/home-assistant/core/pull/34642
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/13220

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [X] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [X] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [X] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [X] 🥇 Gold - I think?
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

MrHarcombe committed:
Converted MetOffice to use UI for configuration
- Sensor and Weather platforms now use config_flow with unique_ids
- MetOfficeData object is now async-protected
- Integration uses DataUpdateCoordinator and so refreshes correctly
- Test cases written to test config_flow and entity behaviours

MrHarcombe committed:
Converted MetOffice to use UI for configuration
- Sensor and Weather platforms now use config_flow with unique_ids
- MetOfficeData object is now async-protected
- Integration uses DataUpdateCoordinator and so refreshes correctly
- Test cases written to test config_flow and entity behaviours

MrHarcombe committed:
Correction to erroneous pylint error, regarding unused DOMAIN import

MrHarcombe committed:
Correction to erroneous pylint error, regarding unused DOMAIN import

MrHarcombe committed:
Added further test cases to improve patch coverage. Now tests more negative cases, as well as the positive cases.

MrHarcombe committed:
Added further test cases to improve patch coverage. Now tests more negative cases, as well as the positive cases.

MatthewFlamm commented on homeassistant/components/metoffice/__init__.py:
```diff
@@ -1 +1,106 @@
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
```
The SCHEMA can probably be removed, you don't use them in `async_setup`

MrHarcombe replied:
True - hadn't spotted that, just followed another's example, but you're right. Done.

MatthewFlamm commented on homeassistant/components/metoffice/__init__.py:
```diff
@@ -1 +1,106 @@
+    name = entry.data[CONF_NAME]
+
+    metoffice_data = MetOfficeData(hass, api_key, latitude, longitude)
+    await metoffice_data.async_update_site()
```
What happens if this fails here but not in the config_flow?  I think `ConfigEntryNotReady` can be raised.  Ideally, we could set up the data inside the config_flow and pass it here, but I couldn't figure that out.

MrHarcombe replied:
That makes sense - I'll give it a go and see what I can do with that.

MrHarcombe replied:
As you say, you can't just pass back the data object (as it can't be easily converted to/from JSON) so I now check the newly created data object for a site name and weather data and raise the exception if either fails.

MatthewFlamm replied:
I was hoping that a core dev would see the redundancy and extra problems here and suggest a good method to handle it in future PRs for both of us

MrHarcombe replied:
Yeah, I looked at whether to try and add methods to serialize the ...Data object thru JSON back into the config_flow handler, but it seemed kinda hokey when all that would do would be to save my code the trouble of actually creating the instance itself (as essentially it would be creating a new instance from the same api_key, latitude and longitude)... overkill? Felt like it.

MatthewFlamm commented on homeassistant/components/metoffice/__init__.py:
```diff
@@ -1 +1,106 @@
+    }
+
+    # Fetch initial data so we have data when entities subscribe
+    await metoffice_coordinator.async_refresh()
```
If any of this data is needed to set up the entity, you probably need to catch an error here too.

MrHarcombe replied:
See above:)

MatthewFlamm commented on homeassistant/components/metoffice/__init__.py:
```diff
@@ -1 +1,106 @@
+    # Fetch initial data so we have data when entities subscribe
+    await metoffice_coordinator.async_refresh()
+
+    if not entry.unique_id:
```
Can this if statement ever register as True?

MrHarcombe replied:
Not sure it can, actually, given that a unique_id gets set in the ConfigEntry during config_flow. Wonder if it can be taken out?

MrHarcombe replied:
Yeah, not sure it can ever be False - what is it achieving?

MatthewFlamm replied:
Shouldn't I be asking you? :wink: The scaffold doesn't have this check, and my own integration doesn't either.

MrHarcombe replied:
Given all that, I've taken it out? Is that safe?

MrHarcombe replied:
I must've gotten the code from somewhere... ah, yes, I got it from digging thru the Coronavirus integration actually. Looking back at that I can see that it was from when I hadn't understood how to properly register a unique_id for an config_entry or an entity - @balloob is doing it in there to convert from an old version of an entity, I think, looking at his code.

MatthewFlamm commented on homeassistant/components/metoffice/sensor.py:
```diff
@@ -30,175 +32,147 @@
-
-# Sensor types are defined like: Name, units
+# Sensor types are defined like: variable -> title, units
+# Sensor types are defined as: name -> title, units, icon
```
Is one of these comments not needed?

MrHarcombe replied:
Correct, and in fact I'm adding an extra column to provide default disabling of the less useful entities so I'll sort that.

MatthewFlamm commented on homeassistant/components/metoffice/sensor.py:
```diff
@@ -30,175 +32,147 @@
 
 class MetOfficeCurrentSensor(Entity):
-    """Implementation of a Met Office current sensor."""
+    """Implementation of a Met Office current weather condition sensor."""
```
Does it make sense to use the available property with the coordinator status?

MrHarcombe replied:
Agreed. Done.

MatthewFlamm commented on homeassistant/components/metoffice/weather.py:
```diff
@@ -1,127 +1,156 @@
-        self.data = data
-        self.site = site
+    def __init__(self, entry_data, hass_data):
+        """Initialise the platform with a data instance."""
```
Same as above, should the available property be set?

MrHarcombe replied:
Agreed. Done.

MatthewFlamm reviewed (commented):
FYI, these comments are based on my experience in writing my own weather integration with config flow.  I do not claim an authoritative understanding.

MrHarcombe committed:
- Code changes following reviewer comments on the PR
- Not all sensor entities are now enabled by default
- Changes to the test cases to only therefore test sensor entities that *are* enabled

MrHarcombe committed:
- Code changes following reviewer comments on the PR
- Not all sensor entities are now enabled by default
- Changes to the test cases to only therefore test sensor entities that *are* enabled

MrHarcombe committed:
- Weather attriute wasn't correctly determining 'available' property
- Extend test coverage by adding a case that interrupts connection

MrHarcombe committed:
- Weather attriute wasn't correctly determining 'available' property
- Extend test coverage by adding a case that interrupts connection

MatthewFlamm commented:
I haven't looked in detail at your tests, but it seems you are testing all the way down to the platforms and entities.  You can probably remove those files from .coveragerc if so.

MrHarcombe commented:
> I haven't looked in detail at your tests, but it seems you are testing all the way down to the platforms and entities. You can probably remove those files from .coveragerc if so.

I am, yes. I had great fun getting to grips with mock requests and fixed date times, to ensure that the data always matched :) - I'll update .coveragerc as you suggest.

MrHarcombe commented:
But as far as I'm concerned it's ready to go?

MartinHjelmare commented on homeassistant/components/metoffice/__init__.py:
```diff
@@ -1 +1,86 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["weather", "sensor"]
```
Please sort :abcd:

MrHarcombe replied:
The platforms? Done.

MartinHjelmare commented on homeassistant/components/metoffice/__init__.py:
```diff
@@ -1 +1,86 @@
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+        if len(hass.data[DOMAIN]) == 0:
```
```suggestion
        if not hass.data[DOMAIN]:
```

MrHarcombe replied:
As you wish - not a problem.

MartinHjelmare commented on homeassistant/components/metoffice/config_flow.py:
```diff
@@ -0,0 +1,78 @@
+            self._abort_if_unique_id_configured()
+            try:
+                info = await validate_input(self.hass, user_input)
+                user_input[CONF_NAME] = info["site_name"]
```
Please only wrap the line(s) that can raise in try... except. Use an `else:` statement if needed.

MartinHjelmare commented on homeassistant/components/metoffice/data.py:
```diff
@@ -0,0 +1,87 @@
+        self.now = None
+
+    @property
+    def latitude(self):
```
We don't need these properties since they are just returning instance attributes. Just use regular instance attributes.

MrHarcombe replied:
Right-o, fair enough.

MartinHjelmare commented on homeassistant/components/metoffice/sensor.py:
```diff
@@ -30,175 +32,163 @@
+        "mdi:weather-sunny",  # but will adapt to current conditions
+        True,
+    ],
+    "temperature": ["Temperature", TEMP_CELSIUS, "mdi:thermometer", True],
```
Temperature sensors should use that device class.

Entities that set a device class should not set a custom icon.

Please set device class for as many entities as possible:
https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

MrHarcombe replied:
Honestly not sure what you mean about this. Where do I set the "device class" in the sensor? Each has name/state/unit/icon attributes - where does the "device class" fit into this, please?

MartinHjelmare replied:
Please implement the `device_class` property in the sensor entity class.

MrHarcombe replied:
Ah right, I see - okay, on it.

MrHarcombe replied:
Done, thanks for that.

MartinHjelmare commented on homeassistant/components/metoffice/sensor.py:
```diff
@@ -30,175 +32,163 @@
+
+        self._type = sensor_type
+        self._name = f"{hass_data[METOFFICE_NAME]} {SENSOR_TYPES[self._type][0]}"
+        self._unique_id = f"{hass_data[METOFFICE_NAME]}_{SENSOR_TYPES[self._type][0]}_{self._data.latitude}_{self._data.longitude}"
```
The configured name should not be part of the unique id. Latitude + longitude + sensor type is enough.

MrHarcombe replied:
No problem.

MartinHjelmare commented on homeassistant/components/metoffice/sensor.py:
```diff
@@ -30,175 +32,163 @@
+        self.metoffice_site_name = self._data.site_name
+        self.metoffice_now = self._data.now
+        self.async_write_ha_state()
+
```
Please also implement `async_update` method and await `coordinator.async_request_refresh`. That will allow users to schedule custom updates via the common entity update service.

MrHarcombe replied:
Done.

MartinHjelmare commented on homeassistant/components/metoffice/strings.json:
```diff
@@ -0,0 +1,22 @@
+        "cannot_connect": "Failed to connect, please try again",
+        "unknown": "Unexpected error"
+      },
+      "abort": { "already_configured": "Device is already configured" }
```
Please use as many common keys as possible. Here are the available common keys:
https://github.com/home-assistant/core/blob/5f8dcd45c1a4279f792f39ef6deffb8f8677cef5/homeassistant/strings.json#L20-L55

Here's an example:
https://github.com/home-assistant/core/blob/5f8dcd45c1a4279f792f39ef6deffb8f8677cef5/homeassistant/components/adguard/strings.json#L7

MrHarcombe replied:
Done.

MrHarcombe replied:
Think that's there, now?

MartinHjelmare commented on homeassistant/components/metoffice/weather.py:
```diff
@@ -1,127 +1,161 @@
+        self._coordinator = hass_data[METOFFICE_COORDINATOR]
+
+        self._name = f"{DEFAULT_NAME} {hass_data[METOFFICE_NAME]}"
+        self._unique_id = f"{DOMAIN}_{self._data.latitude}_{self._data.longitude}"
```
```suggestion
        self._unique_id = f"{self._data.latitude}_{self._data.longitude}"
```

MrHarcombe replied:
Okay.

MartinHjelmare commented on tests/components/metoffice/conftest.py:
```diff
@@ -0,0 +1,26 @@
+from datapoint.exceptions import APIException
+import pytest
+
+from tests.common import mock_coro
```
`mock_coro` is legacy. Please import patch from tests.async_mock. That patch will handle patching both normal functions and coroutines. To mock coroutines explicitly we can use `AsyncMock` if needed.

MrHarcombe replied:
Can't seem to find that import - perhaps my dev branch is out of date. I'll refresh and get back to you...

MrHarcombe replied:
Bother, got the horrible feeling I've messed up my branch (again). Still not particularly expert at this whole Git thing. I shouldn't have merged the latest dev branch onto this branch, should I? It's now re-testing everything as if it was my submission...

MrHarcombe replied:
Okay, Git crisis over (I think) for now. Back onto this tomorrow.

MartinHjelmare commented on tests/components/metoffice/test_config_flow.py:
```diff
@@ -0,0 +1,140 @@
+    ) as mock_setup, patch(
+        "homeassistant.components.metoffice.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
```
Just add a MockConfigEntry for the first entry instead.
https://github.com/home-assistant/core/blob/5f8dcd45c1a4279f792f39ef6deffb8f8677cef5/tests/common.py#L668

MrHarcombe replied:
Think I've done it

MrHarcombe replied:
Nope. I've plainly misunderstood. Could you explain how I should be using the MockConfigEntry, please? I've used one in place of the first entry but it's obviously not what  I was supposed to be doing as now the second entry gets through to "create_entry" instead of "abort"ing?

MrHarcombe replied:
Think I'm there - just having fun and game commiting - something about .pre-commit-hooks.yaml being missing. I'm going to commit straight into GitHub, though, as I've tested it locally.

MartinHjelmare commented on tests/components/metoffice/test_config_flow.py:
```diff
@@ -0,0 +1,140 @@
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text="")
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
```
We can pass the user_input as `data` parameter to `async_init`. Then we don't need to call `async_configure`.

MrHarcombe replied:
Gotcha.

MartinHjelmare commented on tests/components/metoffice/test_metoffice.py:
```diff
@@ -0,0 +1,327 @@
+    all_sites = json.dumps(mock_json["all_sites"])
+    wavertree_hourly = json.dumps(mock_json["wavertree_hourly"])
+
+    from datetime import datetime, timezone
```
Standard library imports should be at the top of the module.

MrHarcombe replied:
They should, yes, but the patch to override "now" doesn't work unless you import datetime at the point you need to mess with it (as per my searching which led to [this document](http://williambert.online/2011/07/how-to-unit-testing-in-django-with-mocking-and-patching/)).

MartinHjelmare replied:
Where is `datetime.datetime` used? This might be a matter of patching in the correct place:
https://docs.python.org/3/library/unittest.mock.html#where-to-patch

MrHarcombe replied:
It's used as the datapoint library checks the date/time and returns the "current" data for the forecast site. I've mocked the API call to feed it some specific data for a given date/time but then need the API to pass me predictable data to test against. Hence the need to mock the date/time.

MartinHjelmare replied:
Have you tried patching datetime where it's used?

MrHarcombe replied:
I did try, but maybe it's my lack of experience with Python mocks, I couldn't get it to work. This was the only way I could seem to get it to behave in a consistent way and mock "now" so that datapoint API returned the expected data.

MartinHjelmare commented on tests/components/metoffice/test_metoffice.py:
```diff
@@ -0,0 +1,327 @@
+    assert len(running_sensor_ids) > 0
+    for running_id in running_sensor_ids:
+        sensor = hass.states.get(running_id)
+        _LOGGER.info(
```
Please remove the logging.

MrHarcombe replied:
Done - was getting myself confused about what to expect, when, hence my logging it. Removed the logger.

MartinHjelmare reviewed (changes requested):
Mostly just some clean up needed. Some changes around sensor with device classes though.

MrHarcombe committed:
Merge branch 'metoffice_config_flow' of https://github.com/MrHarcombe/core into metoffice_config_flow

MrHarcombe commented:
> Did you commit changes without pushing them? Many of my comments have been resolved but have not been addressed in the code.

D'oh - on it.

MrHarcombe committed:
Edits following development review by MartinHjelmare

MrHarcombe committed:
Misunderstood how to replace legacy mock_coro use

MrHarcombe committed:
Using MockConfigEntry (as per developer review) to test re-configuring deployment

MrHarcombe committed:
Correctly using MockConfigEntry

Found an example to work from, then just had to realise to add a proper unique_id for the test location, so the next entry would be refused.

MartinHjelmare commented on homeassistant/components/metoffice/sensor.py:
```diff
@@ -30,175 +34,189 @@
+            ATTR_LAST_UPDATE: self.metoffice_now.date if self.metoffice_now else None,
+            ATTR_SENSOR_ID: self._type,
+            ATTR_SITE_ID: self.metoffice_site_id if self.metoffice_site_id else None,
+            ATTR_SITE_NAME: self.metoffice_site_name
```
Are the site attributes important for automations? We only want to add static info to state attributes if it's essential for automations.

MrHarcombe replied:
I don't use them myself - they were there from previous versions. I'll be honest, all I've done is bring the code in line with how I've updated the data model and UI/config_flow usage. If you don't want them there, then they can go? Which bits don't you want? Any of them?

MrHarcombe replied:
Happy to remove them, to be honest, and see if anyone complains?

MartinHjelmare replied:
Let's keep them for now.

MartinHjelmare commented on tests/components/metoffice/conftest.py:
```diff
@@ -0,0 +1,24 @@
+    """Mock datapoint Manager with default values for testing in config_flow."""
+    with patch("datapoint.Manager") as mock_manager:
+        instance = mock_manager.return_value
+        instance.get_nearest_forecast_site.return_value = AsyncMock(
+            exception=APIException
+        )
```
The `Site` class is just a regular class and not a coroutine. If we want `get_nearest_forecast_site` method to raise an exception we just set that `side_effect`.
```suggestion
        instance.get_nearest_forecast_site.side_effect = APIException()
```

MartinHjelmare commented on tests/components/metoffice/conftest.py:
```diff
@@ -0,0 +1,24 @@
+        instance.get_nearest_forecast_site.return_value = AsyncMock(
+            exception=APIException
+        )
+        instance.get_forecast_for_site.return_value = AsyncMock(exception=APIException)
```
```suggestion
        instance.get_forecast_for_site.side_effect = APIException()
```

MartinHjelmare commented on tests/components/metoffice/test_metoffice.py:
```diff
@@ -0,0 +1,314 @@
+        return datetime.__new__(datetime, *args, **kwargs)
+
+
+@patch("datetime.datetime", MockDateTime)
```
I think we should try to patch something like this instead, but set the patch target path to the library path for datetime.now:
https://github.com/home-assistant/core/blob/3c57475c8f308231c001e816b105d6739a72868c/tests/components/caldav/test_calendar.py#L335

MrHarcombe replied:
Sadly, this doesn't work - if you see in [this pastebin of the test file](https://pastebin.com/WDQm62Gg) I've tried the functions in various combinations of patches, but none work (unless I've missed your point) as you can see [from the output](https://pastebin.com/mrYXakUP).

MrHarcombe replied:
Eureka! Thanks for the assist - did a bit more digging and reading, and found where to patch the datetime override. I'll check it in and hopefully it'll be more what you were looking for :)

MartinHjelmare replied:
Good! I tested this on my end too and this works:
```suggestion
@patch(
    "datapoint.Forecast.datetime.datetime",
    Mock(now=Mock(return_value=datetime(2020, 4, 25, 12, tzinfo=timezone.utc))),
)
```

MrHarcombe replied:
Oh I like how you've got the mock return type in there (I've so much to learn, still...) - I'd not managed to get it that neat. I'll update and re-commit.

MrHarcombe committed:
Update tests/components/metoffice/conftest.py

As per review suggestion

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

MrHarcombe committed:
Update tests/components/metoffice/conftest.py

As per review suggestion

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

MrHarcombe committed:
Removed unused import 

Serves me right for not going through the proper steps...

MrHarcombe committed:
Correct patching of datetime.now

With pointers from @MartinHjelmare, finally found the right way to patching datetime.now so as to be able to freeze time for datapoint API to return known weather data for testing.

MrHarcombe committed:
Neater patching

Thanks to @MartinHjelmare (again) for this very neat way of Mock'ing into datetime.now directly on the patch declaration.

MartinHjelmare commented on homeassistant/components/metoffice/strings.json:
```diff
@@ -0,0 +1,23 @@
+          "api_key": "Met Office DataPoint API key",
+          "latitude": "Latitude",
+          "longitude": "Longitude",
+          "name": "Friendly name"
```
`name` isn't used in the form, right?

MrHarcombe replied:
Correct, I'll remove it.

MartinHjelmare commented on homeassistant/components/metoffice/config_flow.py:
```diff
@@ -0,0 +1,80 @@
+                    CONF_LONGITUDE, default=self.hass.config.longitude
+                ): cv.longitude,
+            },
+            extra=vol.ALLOW_EXTRA,
```
It doesn't look as we need to allow extra items.

MrHarcombe replied:
No. I thought I'd needed it to add the site_name in after connecting to the API, but I've just removed it and re-tested (and run it up locally, to be doubly sure) and it all works fine without. So I'll remove that, too.

MartinHjelmare commented on tests/components/metoffice/test_metoffice.py:
```diff
@@ -0,0 +1,288 @@
+"""The tests for the Met Office weather component."""
```
Please split this module into two modules, `test_sensor.py` and `test_weather.py` and put the corresponding tests in each module.

MrHarcombe replied:
Funnily enough, I did exactly that in the other PR that you marked as draft where I added forecast data and daily update modes!

MrHarcombe replied:
Done.

MrHarcombe committed:
- Removed unused config strings and schema options, as per developer review
- Tweak to icon property

MrHarcombe committed:
Merge branch 'metoffice_config_flow' of https://github.com/MrHarcombe/core into metoffice_config_flow

MrHarcombe committed:
Split testing into separate files for weather and sensor entities (as per review suggestion)

MartinHjelmare reviewed (approved):
Great!

MartinHjelmare commented:
Please add a breaking change paragraph where we briefly describe what changed and what the user needs to do to cope with the breaking change.

MrHarcombe commented:
> Please add a breaking change paragraph where we briefly describe what changed and what the user needs to do to cope with the breaking change.

Done. Thank you!

(How do I go about un-"draft"ing the dependent PR that adds forecast data and daily update mode? Presumably I start by merging in all these updates, then...?)

MartinHjelmare commented:
Yeah, rebase on dev branch after this PR has been merged.

tiems90 commented:
A question here, since I haven't been able to find the answer in the documents. The documents mention the entities that are enabled by default; but how do I enable additional non-default endpoints?

Specifically I am looking for `visibility_distance` and `humidity`. Previously we set these up under `monitored_conditions`, but I cannot find where to set this.

MrHarcombe commented:
Ah, sorry!

If you look in the configuration / integrations / MetOffice you get the
list of current enabled entities...
[image: image.png]

Click on the downward triangle (the filter button) in the top-right of the
view, and you can choose which entities are displayed...
[image: image.png]

From there, you can tick to select the ones you want, then click the button
at the top of the list, to enable the selected entities...
[image: image.png]

Hopefully that should do it for you (on this, and any other integrations
that may start doing this, to you).

--Ian

On Wed, 8 Jul 2020 at 20:24, Timo <notifications@github.com> wrote:

> A question here, since I haven't been able to find the answer in the
> documents. The documents mention the entities that are enabled by default;
> but how do I enable additional non-default endpoints?
>
> Specifically I am looking for visibility_distance and humidity.
> Previously we set these up under monitored_conditions, but I cannot find
> where to set this.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/core/pull/34900#issuecomment-655708886>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQU6A4UXGFHWG4G5K3BHHTR2TBXVANCNFSM4MUEFYYA>
> .
>

tiems90 commented:
Oh wow! Apologies. I had no idea! I was looking through it and it didn’t make sense I saw 13 entities, and not all of them after clicking on them. 

Thanks again for walking me through that; everything is working again! 

> On 8 Jul 2020, at 22:27, Ian Harcombe <notifications@github.com> wrote:
> 
> ﻿
> Ah, sorry!
> 
> If you look in the configuration / integrations / MetOffice you get the
> list of current enabled entities...
> [image: image.png]
> 
> Click on the downward triangle (the filter button) in the top-right of the
> view, and you can choose which entities are displayed...
> [image: image.png]
> 
> From there, you can tick to select the ones you want, then click the button
> at the top of the list, to enable the selected entities...
> [image: image.png]
> 
> Hopefully that should do it for you (on this, and any other integrations
> that may start doing this, to you).
> 
> --Ian
> 
> On Wed, 8 Jul 2020 at 20:24, Timo <notifications@github.com> wrote:
> 
> > A question here, since I haven't been able to find the answer in the
> > documents. The documents mention the entities that are enabled by default;
> > but how do I enable additional non-default endpoints?
> >
> > Specifically I am looking for visibility_distance and humidity.
> > Previously we set these up under monitored_conditions, but I cannot find
> > where to set this.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/home-assistant/core/pull/34900#issuecomment-655708886>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/AAQU6A4UXGFHWG4G5K3BHHTR2TBXVANCNFSM4MUEFYYA>
> > .
> >
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.
